lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -696,3 +696,41 @@ function dosomething_helpers_get_campaign_status($start_date, $end_date) {
     return 'closed';
   }
 }
+
+/**
+ * Gets the current run for a given campaign nid and user.
+ *
+ * @param int $nid
+ * @param object $user
+ *  Optional. Will default to global $user if not supplied
+ *
+ * @return node $run
+ *  Returns the campaigns node in the correct language.
+*/
+function dosomething_helpers_get_campaign_run($nid, $user = NULL) {
```

Comment:
I dont believe `function dosomething_campaign_run_get_closed_run_nid($nid) {` will work under some of the edge cases Fantini has described, which is why I did a lot of seemingly redundant checks in this.